File: src/main/java/com/mercateo/jsonschema/property/RawPropertyCollector.java
package com.mercateo.jsonschema.property;


import com.mercateo.jsonschema.generictype.GenericType;
import javaslang.collection.Stream;

public interface RawPropertyCollector {
    Stream<RawProperty> forType(GenericType<?, ?> genericType);

    default Stream<RawProperty> forType(Class<?> clazz) {
        return forType(GenericType.of(clazz));
    }
}

File: src/main/java/com/mercateo/jsonschema/property/RawPropertyCollector.kt
package com.mercateo.jsonschema.property


import com.mercateo.jsonschema.generictype.GenericType

interface RawPropertyCollector {
    fun <S> forType(genericType: GenericType<S>): Sequence<RawProperty<S, *>>

    fun <S> forType(clazz: Class<S>): Sequence<RawProperty<S, *>> {
        return forType(GenericType.of(clazz))
    }
}
